### PR TITLE
kernel: bump 6.6 to 6.6.33

### PR DESCRIPTION
**Pull request by graysky2**
This commit accomplishes three goals:
1. bump 6.6 to 6.6.33
2. kernel: modules: video: change package definition for fb for upstream changes[1]
3. kernel/multiple subtargets: add CONFIG_FB_IOMEM_FOPS=y to all subtargets which also set CONFIG_FB_CORE=y.

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.33

Removed upstreamed:
	pending-6.6/195-block-fix-and-simplify-blkdevparts-cmdline-parsing.patch[1]
	gemini/patches-6.6/0003-net-ethernet-cortina-Locking-fixes.patch[2]

Manually rebased:
	ramips/patches-6.6/810-uvc-add-iPassion-iP2970-support.patch

All other patches automatically rebased.

@blogic - can you please review my manual rebase of your patch?

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.6.y&id=c00e8fd749502c02085534c60b1edca4fc479c91
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.33&id=99bbbd9aea059f8a206736dc601be2ae61d366fb
3. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.33&id=8f6f82d6a205ceb3aba8d279f9ff6eeea0b1689b

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
Run-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3

### CODE REVIEW

**Review by hauke — APPROVED** [review]

**Comment by john-tho** [comment]
CI failures due to [`FB_IOMEM_CORE`](https://elixir.bootlin.com/linux/v6.6.33/K/ident/CONFIG_FB_IOMEM_FOPS): https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.6.y&id=c00e8fd749502c02085534c60b1edca4fc479c91

https://github.com/openwrt/openwrt/actions/runs/9489954603/job/26152443168
https://github.com/openwrt/openwrt/actions/runs/9489954603/job/26152492471
```
 Package kmod-fb is missing dependencies for the following libraries:
fb_io_fops.ko
make[2]: *** [modules/video.mk:122: /__w/openwrt/openwrt/openwrt/bin/targets/imx/cortexa53/packages/kmod-fb_6.6.33-r1_aarch64_cortex-a53.ipk] Error 1
```

https://github.com/openwrt/openwrt/blob/169a6952802d6f417687df8d04c5414bfda0ae1c/package/kernel/linux/modules/video.mk#L108
_may_ be able to add to `KernelPackage/fb` FILES: `$(LINUX_DIR)/drivers/video/fbdev/core/fb_io_fops.ko@ge6.6`

**Comment by namiltd** [comment]
Missing CONFIG_FB_IOMEM_FOPS=y for some architectures

**Comment by rockdrilla** [comment]
Compile/run-tested: mediatek/filogic (GL.iNet GL-MT6000).

**Comment by john-tho** [comment]
> Missing CONFIG_FB_IOMEM_FOPS for some architectures (imx/cortexa53 and rockchip/armv8)

Adding this to kernel config in those two subtargets is the wrong way to address that problem. They don't build-in any framebuffer options (`CONFIG_FB_CORE=y`), instead they use `FEATURES := display` to build the `kmod-fb` package.

**Comment by graysky2** [comment]
@john-tho - thanks for the great suggestions.  Please see my latest two commits.

File: target/linux/imx/cortexa53/target.mk [hunk 1]
```diff
@@ -2,6 +2,7 @@ ARCH:=aarch64
 BOARDNAME:=NXP i.MX with Cortex-A53 (ARM64)
 CPU_TYPE:=cortex-a53
 KERNELNAME:=Image dtbs
+FEATURES:=display
```
Comment (john-tho) [inline target/linux/imx/cortexa53/target.mk:5]: We don't want to change this, it is already set (at subtarget level): https://github.com/openwrt/openwrt/blob/96978da1d9d83b0bcee67fb8c948daacf8c3092c/target/linux/imx/Makefile#L9 that's why we saw the CI issue. We want to change the package definition for `fb`

**Comment by hauke** [comment]
I think this should all go into one commit as the kernel update needs these changes. Otherwise the versions in between will not build fully.

**Comment by graysky2** [comment]
I can combine into a single commit later today. In the past, I have been encouraged to keep such things as separate commits so your request is a but of a change to that.

**Comment by Borromini** [comment]
Compile-tested and run-tested on mvebu/cortexa72 (RB5009UG+S+IN)

Tested-by: Stijn Segers <foss@volatilesystems.org>

**Comment by namiltd** [comment]
Time for 6.6.34

**Pull request merged**